QuizApp/QuizApp/UI/Screens/SolvingQuiz/ProgressView.swift

```diff
@@ -1,70 +1,24 @@
-import UIKit
-
-class ProgressView: UIView {
-
-    private var label: UILabel!
-    private var stackView: UIStackView!
-
-    override init(frame: CGRect) {
-        super.init(frame: frame)
-
-        createViews()
-        styleViews()
-        defineLayoutForViews()
-    }
-
-    required init?(coder: NSCoder) {
-        fatalError("init(coder:) has not been implemented")
-    }
-
-    func set(colors: [UIColor]) {
-        stackView.arrangedSubviews.forEach { $0.removeFromSuperview() }
-
-        for color in colors {
-            let cell = UIView()
-            cell.backgroundColor = color
-            cell.layer.cornerRadius = 2
-
-            stackView.addArrangedSubview(cell)
+import SwiftUI
+
+struct ProgressView: View {
+
+    let progressText: String
+    let progressColors: [UIColor]
```

Comment:
UIColor is a UIKit class, use Color from SwiftUI. Also, maybe it would be better to define separate struct like
```
struct ProgressData: Identifiable {

   let id: Int // this can be questionId
   let color: Color
}
```

and use it here.